### PR TITLE
Update mask DB schema

### PR DESCRIPTION
**Pull request by chris-miaskowski**
The change I'm proposing is a direct response to the implementation.
Originally, the proposed schema asked for node_id + node_version_id.
However, when during the implementation it turned out that I don't need to store node_id on the mask.
It's a redundancy that can be avoided. 
Basically knowing node_version_id I know node_id.
Keeping track of node_id makes the testing and implementing a little bit more complex and doesn't seem to bring a lot of value to the table. The proposal is to remove it.

To back up my words with code samples here is how I can get a mask w/o keeping node_id store directly:

```
return this.masksRepository.findOne({
      where: {
        dataHash,
        nodeVersion: {
          id,
          node: {
            id: nodeId,
          },
        },
      },
      relations: ['nodeVersion'],
    });
```

### CODE REVIEW

File: docs/model-masking.md [hunk 1]
```diff
@@ -109,12 +106,13 @@ properties:
     description: An md5 hash of the data object for easy equality lookup.
 required: 
   - id
-  - node_id
   - node_version_id
   - data
   - data_hash
 ```
 
+**Note**: node_version_id determines the node_id and therefore there is no need to keep track of the actual node_id.
```
Comment (marbemac) [inline docs/model-masking.md]: It's a denormalization choice that we've made elsewhere to make various queries easier (see node_snapshots have node_id and node_version_id). I suggest that we keep it.

Comment (chris-miaskowski) [reply docs/model-masking.md]: Can you recall any such query that is easier?

**Comment by marbemac** [comment]
> Keeping track of node_id makes the testing and implementing a little bit more complex

How much more complex does a "little bit" mean? :)

Your query example is definitely more complex than:

```
return this.masksRepository.findOne({
      where: {
        dataHash,
        nodeId,
      },
    });
```

In general we've been denormalizing various ids to make queries easier (see node_id on node_snapshots). Or org_id on node_snapshots (technically we could get org id by joining versions, then nodes, then project, then org).

**Comment by marbemac** [comment]
Ah crap I deleted my previous code comment since was basically duplicated in main thread comment, but looks like you might have responded to it already, sorry!

**Comment by chris-miaskowski** [comment]
Hey @marbemac, let's close this thread. The reduce of complexity is negligible - probably this thread took longer than the actual implementation.

**Comment by chris-miaskowski** [comment]
Thanks for the feedback though!

**Comment by marbemac** [comment]
Agree :)

**Pull request closed without merging**